gdal2mbtiles/vips.py

```diff
@@ -188,7 +116,11 @@ def set_concurrency(self, processes):
 VIPS = LibVips()
 
 
-class VImage(vipsCC.VImage.VImage):
+class VImageAdapter(object):
```

Comment:
Trying to wrap my head around this new vips library here, so bear with me; what's the rationale for not inheriting from `pyvips.Image` (the pyvips equivalent of `vipsCC.Vimage.VImage` AFAICT)?

Reply:
I tried to, but ran into issues with the object that was returned from certain methods.  Mostly because if you try to call a pyvips.Image operation that hasn't been explicitly defined, it tries to look it up in the underlying library, and then returned a pyvips.Image rather than an inherited VImage.  (I think, although I'm slightly vague now about what the exact error was now)

Reply:
Fair enough. That would just be the same behaviour as before though, so I guess I'm just surprised that it's an issue with pyvips.

Reply:
@mghughes Just out of interest, I went back to this to see if I could subclass `pyvips.Image` after all, using a `from_vimage` classmethod as we did before.  However, the issue is that `pyvips.Image` is instantiated with a pointer.  Trying to instantiate a new class with the other.pointer as below results in all sorts of errors of the sort:
```
GLib-GObject-WARNING **: invalid unclassed type '<invalid>' in class cast to 'GObject'
GLib-GObject-CRITICAL **: g_object_class_find_property: assertion 'G_IS_OBJECT_CLASS (class)' failed
```
It did seem to kind of work, despite the warnings/errors, but it couldn't find readonly properties from `pyvips.Image` on the child class (e.g. `width`).  Removing the `new.__dict__.update(other.__dict__)` line let it find properties, but failed in various other ways.

```
@classmethod
def from_vimage(cls, other):
    new = cls(other.pointer)
    new.__dict__.update(other.__dict__)
    return new
```

I'm giving up on this attempt now :)